### DISCUSSION TITLE
Triggering DROP when DTR matches that in a background profile

### DISCUSSION THREAD

**Discussion by bowerhaus** [artisan-roaster-scope/artisan#1206] [Q&A] [answered]
Is there a way of triggering a DROP when the current DTR in the roast matches the DTR at which the DROP occurred in a background profile? I know that in the Background dialog I can set a DROP to occurs when the BT or the time matches that in the background but I can't see a way of doing this for the Development Ratio?

There seems to be something where I might be able to create a Symbolic Formula and then trigger an Alarm off that. This is a bit new to me but, if I use this, it looks like I would have to set the target DTR as a constant whereas I want it to be driven by the background profile.

Thanks

**Comment by roasterdave** [comment, marked as answer]
Yes, this is possible using a Virtual Device with a symbolic equation to compute the difference between the current roast's DTR and the background DTR and an alarm to trigger DROP when the current roast's DTR matches the background DTR.

First, create a new **Virtual Device** (Config>> Device>> Extra Devices tab).  Replace the **Label1** "Extra 1" with something like "DTRdiff".  Then copy and paste this equation into the **y1(x)** field.
`100*(((t-t{FCs})/(t-t{CHARGE}))-((b{DROP}-b{FCs})/(b{DROP}-b{CHARGE})))`

If you want to observe this value check **LCD1**.  Uncheck the rest of the options.  Be aware, the LCD will show "uu" until **FCs** is marked in the current roast.

Now create a new alarm (Config>> Alarms).  Set the alarm **Source** to the virtual device created above. Set the alarm **Condition** to "above", set the **Value** to "0.0" and set the **Action** to "DROP".  For a variety of reasons the DTRs at DROP may not match exactly to the decimal place.

- Please note that due to a problem reporting background events to symbolic equations found in v2.8.4 you must use the [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous).  Future releases will have this this correction.

![1206-1](https://github.com/artisan-roaster-scope/artisan/assets/30009312/9f908803-1063-42fd-a368-ddf2437e0038)
![1206-2](https://github.com/artisan-roaster-scope/artisan/assets/30009312/4edff9c7-66d2-4ab9-a749-0245dbae9ab1)

**Comment by Terracotta-62** [comment]
Good idea It has been proven to work very well.

**Reply by bowerhaus** [reply]
Cool - sounds just what I'm looking for - let me try it tonight, thank you.

Can you tell me if the bug you mention in 2.8.4 might be responsible for something I'm seeing. In one setup I have I see that Background events are only distributed based on temperature even though the time option is selected. In another setup I don't see background events being triggered at all (and Playback is definitely turned on). Thanks.

**Reply by MAKOMO** [reply]
The v2.8.4 issue is definitely not related to the background event processing. If you see an issue there that you would like to be fixed, please file a separate Issue providing all the necessary information (like Artisan settings and profiles) that allow us to reproduce the issue.

Note that before the TP is recognized replay is always by time.

**Reply by bowerhaus** [reply]
Ok, thanks, I'll look into the problem a bit more before filing an issue.

**Reply by bowerhaus** [reply]
Hi again. I see that the continuous build does not yet include an RPi install. I'm currently using BullsEye and 2.8.2 of Artisan so what is the best route to upgrading to the latest version as things stand now?

**Reply by MAKOMO** [reply]
There is no build for the current v2.8.4 release nor for any continuous build. As we write on the v2.8.4 download page, we will build RPi packages for RPi OS bookworm once it is available. There will not be a version beyond 2.8.2 for any previous version of RPi OS. We are looking for an RPi build maintainer!

**Reply by bowerhaus** [reply]
Ah okay. Looks like I'll have to wait a bit before trying the suggestion to my original post.

**Reply by bowerhaus** [reply]
Hi, I've now switched over to Windows from RPi, and have loaded up the continuous build. Using this @roasterdave 's reply works really well. Many thanks.
